### PR TITLE
A4A: Show an 'Early Access' banner on the Agency Tier and benefits page.

### PR DESCRIPTION
**Pull request by jkguidaven**
This PR adds a disclaimer on the Agency Tier and benefits page that clarifies the early access experience.
Screenshot 2024-11-13 at 4 22 10 PM

Closes https://github.com/Automattic/automattic-for-agencies-dev/issues/1468

## Proposed Changes

* Create an `EarlyAccessBanner` component based from the `LayoutBanner` component. This component can be dismissed, and its dismiss state will be saved as a preference.

## Why are these changes being made?

* This is part of the Agency Tier project and needed before the launch.

## Testing Instructions

* Use the A4A live link and go to the `/agency-tier` page.
* Please confirm if the banner is visible. If it isn’t, reset the `a4a-agency-tier-early-access-banner-dismissed` preference with the Calypso debug tool.

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

File: client/a8c-for-agencies/sections/agency-tier/early-access-banner/index.tsx [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+	const translate = useTranslate();
+	const dispatch = useDispatch();
+
+	const bannerDismissed = useSelector( ( state ) =>
+		getPreference( state, EARLY_ACCESS_BANNER_DISMISSED )
+	);
+
+	const handleDismiss = () => {
+		dispatch( savePreference( EARLY_ACCESS_BANNER_DISMISSED, true ) );
+		dispatch( recordTracksEvent( 'calypso_a4a_early_access_banner_dismissed' ) );
```
Comment (yashwin) [inline client/a8c-for-agencies/sections/agency-tier/early-access-banner/index.tsx]: Do you think we should update the event name?

```suggestion
		dispatch( recordTracksEvent( 'calypso_a4a_agency_tier_early_access_banner_dismissed' ) );
```

**Review by yashwin — APPROVED** [review]
LGTM!

**Review by simonktnga8c — APPROVED** [review]
Nice work @jkguidaven 🥳

File: client/a8c-for-agencies/sections/agency-tier/early-access-banner/index.tsx (hunk 1 above)
Comment (jkguidaven) [reply client/a8c-for-agencies/sections/agency-tier/early-access-banner/index.tsx]: Thanks for the review, I've renamed it to `calypso_a4a_agency_tier_early_access_banner_dismiss_click`

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/17001835

Some locales (Hebrew, Japanese) have been temporarily machine-translated due to translator availability. All other translations are usually ready within a few days. Untranslated and machine-translated strings will be sent for translation next Monday and are expected to be completed by the following Friday.

Thank you @jkguidaven for including a screenshot in the description! This is really helpful for our translators.

**Pull request merged**